src/new_index/db_metrics.rs

```diff
@@ -0,0 +1,225 @@
+use crate::metrics::{Metrics, MetricOpts, GaugeVec};
+
+#[derive(Debug)]
+pub struct RocksDbMetrics {
+    // Memory table metrics
+    pub num_immutable_mem_table: GaugeVec,
+    pub mem_table_flush_pending: GaugeVec,
+    pub cur_size_active_mem_table: GaugeVec,
+    pub cur_size_all_mem_tables: GaugeVec,
+    pub size_all_mem_tables: GaugeVec,
+    pub num_entries_active_mem_table: GaugeVec,
+    pub num_entries_imm_mem_tables: GaugeVec,
+    pub num_deletes_active_mem_table: GaugeVec,
+    pub num_deletes_imm_mem_tables: GaugeVec,
+
+    // Compaction metrics
+    pub compaction_pending: GaugeVec,
+    pub estimate_pending_compaction_bytes: GaugeVec,
+    pub num_running_compactions: GaugeVec,
+    pub num_running_flushes: GaugeVec,
+
+    // Error metrics
+    pub background_errors: GaugeVec,
+
+    // Key and data size estimates
+    pub estimate_num_keys: GaugeVec,
+    pub estimate_live_data_size: GaugeVec,
+    pub estimate_oldest_key_time: GaugeVec,
+
+    // Table reader memory
+    pub estimate_table_readers_mem: GaugeVec,
+
+    // File and SST metrics
+    pub is_file_deletions_enabled: GaugeVec,
+    pub total_sst_files_size: GaugeVec,
+    pub live_sst_files_size: GaugeVec,
+    pub min_obsolete_sst_number_to_keep: GaugeVec,
+
+    // Snapshot metrics
+    pub num_snapshots: GaugeVec,
+    pub oldest_snapshot_time: GaugeVec,
+
+    // Version metrics
+    pub num_live_versions: GaugeVec,
+    pub current_super_version_number: GaugeVec,
+
+    // Log metrics
+    pub min_log_number_to_keep: GaugeVec,
+
+    // Level metrics
+    pub base_level: GaugeVec,
+
+    // Write metrics
+    pub actual_delayed_write_rate: GaugeVec,
+    pub is_write_stopped: GaugeVec,
+
+    // Block cache metrics
+    pub block_cache_capacity: GaugeVec,
+    pub block_cache_usage: GaugeVec,
+    pub block_cache_pinned_usage: GaugeVec,
+}
+
+impl RocksDbMetrics {
+    pub fn new(metrics: &Metrics) -> Self {
+        let labels = &["rocksdb"];
+
+        Self {
+            // Memory table metrics
```

Comment:
some of these help messages look generic, we should be copying the comments verbatim from https://github.com/facebook/rocksdb/blob/main/include/rocksdb/db.h